plugins/topology/src/components/Graph/WorkloadNode.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+import {
+  GraphElement,
+  isNode,
+  Node,
+  NodeStatus,
+  observer,
+  ScaleDetailsLevel,
+  useHover,
+  useVisualizationController,
+  WithDragNodeProps,
+  WithSelectionProps,
+} from '@patternfly/react-topology';
+import BaseNode from './BaseNode';
+import PodSet, { podSetInnerRadius } from '../Pods/PodSet';
+import { AllPodStatus } from '../Pods/pod';
+import { calculateRadius, getPodStatus } from '../../utils/workload-node-utils';
+import { getNodeDecorators } from '../../utils/getWorkloadNodeDecorators';
+import { getTopologyResourceObject } from '../../utils/topology-utils';
+import { usePodsWatcher } from '../../hooks/usePodsWatcher';
+import { K8sResourceKind } from '../../types/types';
+
+import './WorkloadNode.css';
+
+const POD_STATUS_NORMAL = 1;
+const POD_STATUS_WARNING = 2;
+const POD_STATUS_DANGER = 3;
+
+const StatusSeverities = {
+  danger: [
+    'ContainerCannotRun',
+    'CrashLoopBackOff',
+    'Critical',
+    'ErrImagePull',
+    'Error',
+    'Failed',
+    'Failure',
+    'ImagePullBackOff',
+    'InstallCheckFailed',
+    'Lost',
+    'Rejected',
+    'UpgradeFailed',
+  ],
+  warning: [
+    'Cancelled',
+    'Deleting',
+    'Expired',
+    'Not Ready',
+    'Terminating',
+    'Warning',
+    'RequiresApproval',
+  ],
+  normal: [
+    'New',
+    'Pending',
+    'Planning',
+    'ContainerCreating',
+    'UpgradePending',
+    'In Progress',
+    'Installing',
+    'InstallReady',
+    'Replacing',
+    'Running',
+    'Updating',
+    'Upgrading',
+    'Accepted',
+    'Active',
+    'Bound',
+    'Complete',
+    'Completed',
+    'Created',
+    'Enabled',
+    'Succeeded',
+    'Ready',
+    'Up to date',
+    'Provisioned as node',
+    'Preferred',
+    'Connected',
+    'Info',
+    'Unknown',
+    'PipelineNotStarted',
+  ],
+};
+
+const getNodePodStatus = (podStatus: AllPodStatus): number => {
+  switch (podStatus) {
+    case AllPodStatus.Failed:
+    case AllPodStatus.CrashLoopBackOff:
```

Comment:
Have added `ErrImagePull` as well here. 

<img width="355" alt="image" src="https://user-images.githubusercontent.com/5129024/220551351-b20e5502-5860-47b9-8c60-5a7b55805bd9.png">


In podStatus for topology I don't see this being handled in the console as there in topology, error in podrings will be shown only pod status is failed and for errImagePull status will be in Pending

```
...
status:
  phase: Pending
  ...
  containerStatuses:
    - name: container
      state:
        waiting:
          reason: ImagePullBackOff
          message: Back-off pulling image "jai/helloworld56"

```
and the best reasoning I can think of is there we have git import flows where we do s2i or pipeline builds so until the build is complete the image won't be accessible so the user will see a pending state over failed.

Reply:
There are a number of ways this can occur (typo in image name, private image that can't be pulled, image registry is down, etc). We probably want to handle `ImagePullBackOff` as well.

Reply:
Yeah, I guess I can see the argument that the build hasn't completed yet, though, in which case we want pending. Maybe we should check to see if it wasn't intentional.

Reply:
currently, it'll show pending only if phase is pending with the exception to `CrashLoopBackOff` ,  `ImagePullBackOff ` or state is terminated with `exitCode != 0` in containerStatuses state

 https://github.com/janus-idp/backstage-plugins/pull/104/files#diff-8da95685163a1fdc3228145eccfedce06be50826d3d5f2e64d135106c5a02b21R65-R76